pgmpy/metrics/metrics.py

```diff
@@ -394,3 +395,74 @@ def fisher_c(model, data, ci_test, show_progress=True):
     C = -2 * np.log(cis.loc[:, "p_value"]).sum()
     p_value = 1 - stats.chi2.cdf(C, df=2 * cis.shape[0])
     return p_value
+
+
+def SHD(true_model, est_model):
+    """
+    Computes the Structural Hamming Distance between `true_model` and `est_model`.
+
+    SHD is defined as total number of basic operations: adding edges, removing
+    edges, and reversing edges required to transform one graph to the other. It
+    is a symmetrical measure.
+
+    The code first accounts for edges that need to be deleted (from true_model),
+    added (to true_model) and finally edges that need to be reversed. All operations
+    count as 1.
+
+    Parameters
+    ----------
+    true_model: pgmpy.base.DAG or pgmpy.base.CPDAG or pgmpy.models.BayesianNetwork
+        The first model to compare.
+    est_model: pgmpy.base.DAG or pgmpy.base.CPDAG or pgmpy.models.BayesianNetwork
+        The second model to compare.
+
+    Returns
+    -------
+    int:
+        If both true_model and est_model are DAGs or Bayesian Networks returns
+        an integer.
+
+    Examples
+    --------
+    >>> from pgmpy.metrics import SHD
+    >>> from pgmpy.models import BayesianNetwork
+    >>> dag1 = BayesianNetwork([(1, 2), (2, 3)])
+    >>> dag2 = BayesianNetwork([(2, 1), (2, 3)])
+    >>> SHD(dag1, dag2)
+    1
+    """
+    if set(true_model.nodes()) != set(est_model.nodes()):
+        raise ValueError("The graphs must have the same nodes.")
+
+    nodes_list = true_model.nodes()
+
+    dag_true = nx.DiGraph(true_model.edges())
+    adj_mat_true = nx.adjacency_matrix(dag_true, nodelist=nodes_list).todense()
+
+    dag_est = nx.DiGraph(est_model.edges())
+    adj_mat_est = nx.adjacency_matrix(dag_est, nodelist=nodes_list).todense()
+
+    shd = 0
+
+    m1 = adj_mat_true
+    m2 = adj_mat_est
```

Comment:
I don't think these two lines add anything. Why not just directly assign m1 and m2 in lines 440 and 443?

Reply:
True, my bad.